src/Future.re

```diff
@@ -21,21 +51,23 @@ let make = resolver => {
   Future(
     resolve =>
       switch (data^) {
-      | Some(result) => resolve(result)
+      | Some(result) => trampoline(() => resolve(result))
```

Comment:
Doesn't this mean the trampoline still runs on every invocation?

Reply:
Yes my mistake, will fix in next commit.